docs/email/postfix/postfix-spf-dkim-debian-8.md

```diff
@@ -0,0 +1,401 @@
+
+### Adding the SPF policy agent to Postfix
+
+The Python SPF policy agent adds SPF policy checking to Postfix. The SPF record for the sender's domain for incoming mail will be checked and if it exists mail will be handled accordingly. There's a Perl version of the policy agent, but it lacks the full capabilities of the Python one.
+
+1.  If you are using SpamAssassin to do spam filtering, you may want to edit `/etc/postfix-policyd-spf-python/policyd-spf.conf` to change the `HELO_reject` and `Mail_From_reject` settings to `False`. That will cause the SPF policy agent to run it's tests and and add a message header with the results in it but _not_ reject any messages. You may also want to make this change if you want to see the results of the checks but not actual apply them to mail processing. Otherwise, just go with the standard settings.
+
+2.  Edit `/etc/postfix/master.cf` and add the following entry at the end:
+
+    {: .file-excerpt}
+    /etc/postfix/master.cf
+    :   ~~~ text
+        policyd-spf  unix  -       n       n       -       0       spawn
+            user=policyd-spf argv=/usr/bin/policyd-spf
+        ~~~
+
+3.  Open `/etc/postfix/main.cf` and add this entry to increase the Postfix policy agent timeout, to prevent Postfix from aborting the agent if transactions are running a bit slowly:
+
+    {: .file-excerpt}
+    /etc/postfix/main.cf
+    :   ~~~ conf
+        policyd-spf_time_limit = 3600
+        ~~~
+
+4.  Edit the `smtpd_recipient_restrictions` entry to add a `check_policy_service` entry thusly:
+
+    {: .file-excerpt}
+    /etc/postfix/main.cf
+    :   ~~~ conf
+        smtpd_recipient_restrictions =
+            ...
+            reject_unauth_destination,
+            check_policy_service unix:private/policyd-spf,
+            ...
+        ~~~
+
+    Make sure to add the `check_policy_service` entry **after** the `reject_unauth_destination` entry to avoid having your system become an open relay. If `reject_unauth_destination` is the last item in your restrictions list, add the comma after it and omit the comma at the end of the `check_policy_service` item above.
+
+5.  Restart Postfix with:
+
+        systemctl restart postfix
+
+You can check the operation of the policy agent by looking at raw headers on incoming mail messages for the SPF results header. You'll find errors logged in `/var/log/mail.log`. The header the policy agent adds to messages should look something like this:
+
+    Received-SPF: Pass (sender SPF authorized) identity=mailfrom; client-ip=127.0.0.1; helo=mail.example.com; envelope-from=text@example.com; receiver=tknarr@silverglass.org
```

Comment:
The closest to this I have in my logs is:
```
policyd-spf[11571]: Pass; identity=mailfrom;
```

Is this another version of the same message?